### ISSUE TITLE
Issue with Terraform SOPS provider, PGP key and Terraform Cloud running on custom terraform cloud agents

### ISSUE THREAD

**Issue by ursuad** [carlpett/terraform-provider-sops#62]
The SOPS Terraform provider doesn't seem to run correctly on the Terraform Cloud Agents, fails during refresh phase, before the actual plan. 
Versions of tools used:

```
• terraform 0.13.4 
• terraform-provider-sops v0.6.0
• gpg (GnuPG) 2.2.19
• libgcrypt 1.8.5
```

I have custom Docker images running as TFC Agents, so I can install/customise tools to debug this further. During Docker build of those images, I import a secret PGP key used for the encryption of the secrets in the keychain using `gpg --import`

Whenever I run the Terraform Cloud workspace, it first tries to do a `plan` (and as a part of that, because it's terraform 0.13.4, it tries to do a refresh of the data sources). That step fails with the following obscure error that doesn't offer enough information about what's specifically failing:

```
Terraform v0.13.4
Initializing plugins and modules...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.sops_file.gitlab_secrets_sops: Refreshing state... [id=-]
data.sops_file.cloudflare_secrets_sops: Refreshing state... [id=-]
data.sops_file.yandex_secrets_sops: Refreshing state... [id=-]
data.sops_file.datadog_secrets_sops: Refreshing state... [id=-]
data.sops_file.fluxcd_secrets_sops: Refreshing state... [id=-]
module.rc_db.random_password.pwd: Refreshing state... [id=none]

Error: Error getting data key: 0 successful groups required, got 0

  on data.tf line 8, in data "sops_file" "fluxcd_secrets_sops":
   8: data "sops_file" "fluxcd_secrets_sops" {

Error: Error getting data key: 0 successful groups required, got 0

  on data.tf line 12, in data "sops_file" "gitlab_secrets_sops":
  12: data "sops_file" "gitlab_secrets_sops" {

```
Dropping in a session on the agent, and installing SOPS, I can decrypt the secret files manually using `sops -d` just fine, so the right PGP key seems to be imported correctly.
 
I wonder why during the plan/refresh is failing but running `sops -d` manually on the agent works. 🤔 

I'm running out of ideas, is there any way to enable the `sops --verbose` logging option for SOPS using the Terraform Sops Provider, or are there any other things I could try to debug this, considering the logs are not very detailed/helpful with regards to what's failing? 

I tried also set the `TF_LOG` to `DEBUG`  but this didn't help for sops provider. 

Thanks for the help.

**Comment by carlpett** [comment]
Hey @ursuad,
That somewhat unhelpful error message basically means no suitable decryption key was available. So the question then is what differentiates your interactive session from the TF Cloud one. Is there a GPG agent involved?

**Comment by ursuad** [comment]
Hi @carlpett 👋 , thanks for replying so soon. 
I'm just installing the `gnupg` project on the agent with the default setup. 

```
apt-get -y update && apt-get -y install  gnupg
```
Could there be something around having a terminal open in my manual decryption process which makes it work ?

**Comment by carlpett** [comment]
Hey,
Follow-up took a bit longer, sorry. The agent likely needs to get started, which I think might happen, gated by the session being interactive, in a .bashrc or similar.
You could perhaps try importing the key as part of the run instead of during the build? That should ensure the agent is started, and is approximately how we do in the integration tests in this repo.

**Comment by ursuad** [comment]
Hi, I can see that version [3.7 of SOPS ](https://github.com/mozilla/sops/releases/tag/v3.7.0) added support for 
```
On failed decryption with GPG, return the error returned by GPG to the sops user
```
Would it be possible to get a new version of the SOPS provider which relies on the 3.7 version of SOPS? 
Hopefully that should give us some more information.

**Comment by ursuad** [comment]
Thanks for the feedback, I'll give it a try to import the keys as a part of the run as well. I didn't want to do this initially, as it will use local-exec and every time it will show me that  there are changes, which is not ideal.

**Comment by carlpett** [comment]
Oh, nice. If you feel up to it a PR to bump sops would be great, otherwise I'll try to get it done as soon as I can

**Comment by dgem** [comment]
Hi @ursuad et al. 

**tldr;** add an environment var `GNUPGHOME` to the TF workspace set to `/root/.gnupg`

The problem down the the home directory, which is set at runtime to something like `/root/.tfc-agent/component/terraform/runs/run-blablablabla/config/`

If you add the following TF you can see what's happening and how to fix it:
```
resource "null_resource" "gpg" {
  triggers = {
    always_run = "${timestamp()}"
  }
  provisioner "local-exec" {
    command = "gpg -k && GNUPGHOME=\"/root/.gnupg\" gpg -k"
  }
}
```
The solution is to add an environment var `GNUPGHOME` to the TF workspace and point it to `/root/.gnupg` that way it will find the keys imported during the docker build.

having said that... a bump of the sops provider will still be nice ;)

**Comment by carlpett** [comment]
Thanks @dgem!
I have a bit of time this morning, so I'll hopefully have a new version out soon.
